### DISCUSSION TITLE
Train Data, Test Data, Hold-Out Data, Unseen Data: Still Confused :-(

### DISCUSSION THREAD

**Discussion by tonpimenta** [pycaret/pycaret#3856] [Q&A]
Hello Community,
I'm trying to find a clear definition for this, but I'm struggling.
`Setup()` splits into **train** and **test data**. 
- Is the **test data** unseen during the `compare_models()` processes? So the metrics obtained with `compare_models()` have nothing to do with the **test data**? Are these metrics (from `compare_models()`) related to the splitting and shuffling resulting from the k fold cross validation?
- Is the **test data** the so called **"hold-out data"** used for model evaluation and `predict()`? 
- the `evaluate_model()` and `plot_model()` results are based on **test data**, which is **unseen** during training?
- Sometimes I see examples where people splits data into two datasets into "unseen" and "train" **BEFORE** running `setup()` and still using `test_data=None` within `setup()`. Then using `predict()` on this "unseen". I don't quite understand why... considering that we already have another "unseen", which would be the test data created by `setup()`.

**Comment by celestinoxp** [comment]
I believe the confusion started after version 2, version 3 brought some changes that may initially be confusing.
- Make sure you are using the latest version 3.2.0
- in setup the data is internally divided into training and testing, however, when we use the finalize_model command, the data is trained in a single dataset. When using predict_model you must use data not used in the initial dataset.

Example (https://pycaret.readthedocs.io/en/latest/index.html):

from pycaret.datasets import get_data
juice = get_data('juice')
from pycaret.classification import *
exp_name = setup(data = juice,  target = 'Purchase')
lr = create_model('lr')
pred_holdout = predict_model(lr)
pred_unseen = predict_model(lr, data = unseen_dataframe)

**Comment by samlakhmani** [comment]
Still not clear. 

What is unseen data? 

Lets say I have 100,000 rows

I perform train-test-split of 20-80

- 100,000
   - 20,000 : this is unseen data ? 
   - 80,000 : I put this in the setup, where I have another train and holdout split which is 0.7 default. Thus it keeps 30*80,000 = 24,000 for holdout. And my model is trained on the remaining 56,000
      
Now when I run :  
> predict_model(lr)

from the below code, I am essentially getting a predictions for the 24,000 users 

  
```
from pycaret.datasets import get_data
juice = get_data('juice')
from pycaret.classification import *
exp_name = setup(data = juice, target = 'Purchase')
lr = create_model('lr')
pred_holdout = predict_model(lr)
pred_unseen = predict_model(lr, data = unseen_dataframe)
```

**Reply by KazRyding** [reply]
Unseen data is new data that you would like a prediction on.
For example, lets say you trained and saved a model in December. Now say at end of January you want to load December’s model and feed it January’s weather data to predict into Feb without having to retrain a model.
